source/Files.cpp

```diff
@@ -148,352 +131,196 @@ void Files::Init(const char * const *argv)
-			list.push_back(directory + Utf8::ToUTF8(ffd.cFileName) + '/');
-	} while(FindNextFileW(hFind, &ffd));
-
-	FindClose(hFind);
-#else
-	DIR *dir = opendir(directory.c_str());
-	if(!dir)
-		return list;
-
-	while(true)
-	{
-		dirent *ent = readdir(dir);
-		if(!ent)
-			break;
-		// Skip dotfiles (including "." and "..").
-		if(ent->d_name[0] == '.')
-			continue;
-
-		string name = directory + ent->d_name;
-		// Don't assume that this operating system's implementation of dirent
-		// includes the t_type field; in particular, on Windows it will not.
-		struct stat buf;
-		stat(name.c_str(), &buf);
-		bool isDirectory = S_ISDIR(buf.st_mode);
-
-		if(isDirectory)
-		{
-			if(name.back() != '/')
-				name += '/';
-			list.push_back(name);
-		}
-	}
-
-	closedir(dir);
-#endif
-
-	sort(list.begin(), list.end());
-	return list;
-}
-
-
+	for(const auto &entry : filesystem::directory_iterator(directory))
+		if(entry.is_directory())
+			list.emplace_back(entry);
 
-vector<string> Files::RecursiveList(const string &directory)
-{
-	vector<string> list;
-	RecursiveList(directory, &list);
 	sort(list.begin(), list.end());
 	return list;
 }
 
 
 
-void Files::RecursiveList(string directory, vector<string> *list)
+vector<filesystem::path> Files::RecursiveList(const filesystem::path &directory)
 {
-	if(directory.empty() || directory.back() != '/')
-		directory += '/';
-
-#if defined _WIN32
-	WIN32_FIND_DATAW ffd;
-	HANDLE hFind = FindFirstFileW(Utf8::ToUTF16(directory + '*').c_str(), &ffd);
-	if(hFind == INVALID_HANDLE_VALUE)
-		return;
-
-	do {
-		if(ffd.cFileName[0] == '.')
-			continue;
-
-		if(!(ffd.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY))
-			list->push_back(directory + Utf8::ToUTF8(ffd.cFileName));
-		else
-			RecursiveList(directory + Utf8::ToUTF8(ffd.cFileName) + '/', list);
-	} while(FindNextFileW(hFind, &ffd));
-
-	FindClose(hFind);
-#else
-	DIR *dir = opendir(directory.c_str());
-	if(!dir)
-		return;
+	vector<filesystem::path> list;
+	if(!Exists(directory) || !is_directory(directory))
+		return list;
 
-	while(true)
-	{
-		dirent *ent = readdir(dir);
-		if(!ent)
-			break;
-		// Skip dotfiles (including "." and "..").
-		if(ent->d_name[0] == '.')
-			continue;
-
-		string name = directory + ent->d_name;
-		// Don't assume that this operating system's implementation of dirent
-		// includes the t_type field; in particular, on Windows it will not.
-		struct stat buf;
-		stat(name.c_str(), &buf);
-		bool isRegularFile = S_ISREG(buf.st_mode);
-		bool isDirectory = S_ISDIR(buf.st_mode);
-
-		if(isRegularFile)
-			list->push_back(name);
-		else if(isDirectory)
-			RecursiveList(name + '/', list);
-	}
+	for(const auto &entry : filesystem::recursive_directory_iterator(directory))
+		if(entry.is_regular_file())
+			list.emplace_back(entry);
 
-	closedir(dir);
-#endif
+	sort(list.begin(), list.end());
+	return list;
 }
 
 
 
-bool Files::Exists(const string &filePath)
+bool Files::Exists(const filesystem::path &filePath)
```

Comment:
Do we still need this function?

Reply:
It helps to minimize the number of changes, especially for other pull requests using this function.